bin/k8s-deploy

```diff
@@ -25,6 +27,46 @@ done
 echo "Done deploying Secrets"
 echo ""
 
+echo "Deploying Object Store Secrets"
+for OBJSTORE_SECRET_FILE in "${OBJSTORE_SECRET_FILES[@]}"
+do
+  # Make a place to put the files
+  SECRETS_PATH=$(mktemp -d)
+  trap "rm -rf ${SECRETS_PATH}" EXIT
+
+  # Use the filename as the secret name
+  SECRET_NAME=${OBJSTORE_SECRET_FILE##*/}
+  SECRET_NAME=${SECRET_NAME%.secret.objecturl}
+
+  OBJSTORE_URL="$(cat ${OBJSTORE_SECRET_FILE})"
+  if [[ "$OBJSTORE_URL" == s3* ]]; then
+    OBJSTORE_UTIL="aws s3 cp --recursive"
+  else
+    OBJSTORE_UTIL="gsutil rsync -r"
+  fi
+
+  ${OBJSTORE_UTIL} "${OBJSTORE_URL}" "${SECRETS_PATH}"
+  kubectl create secret \
+    generic "${SECRET_NAME}" \
+    --from-file "${SECRETS_PATH}" \
+    --dry-run -o yaml | kubectl apply "--namespace=${NAMESPACE}" -f -
+
+  # Remove local copy
+  rm -rf ${SECRETS_PATH}
+done
+echo "Done deploying Object Store Secrets"
+echo ""
+
+echo "Deploying Encrypted Secrets"
+for SOPS_SECRET_FILE in "${SOPS_SECRET_FILES[@]}"
+do
+  set -x
+  sops ${SOPS_OPTIONS} --decrypt "${SOPS_SECRET_FILE}" | \
+  kubectl apply "--namespace=${NAMESPACE}" -f -
+done
+echo "Done deploying Encrypted Secrets"
+echo ""
+
```

Comment:
I'm not sure I want to try to optimize for catching one specific kind of error (e.g. cloud api call failing) but not catch others (e.g. configmap manifest isn't valid). Part of the value I see in doing it this way is that secrets are treated consistently with all other resources, and all secrets are treated consistently with each other, too. I'd rather have consistent, unsurprising behavior of "errors during deployment _will_ result in partial deployment" rather than "errors during deployment _may_ result in partial deployment in some cases".

Reply:
I guess I disagree with that. If we have a chance to not partially deploy things, I'd rather that be the case. kubectl (last I checked) eventually can validate files before deploy (open issue requiring refactoring) and I'd like to do that for syntax checks at least. There are many ways for calls to external services that aren't kube to fail and I think it would be good to limit how those can impact deployed systems.